ui/app/components/app/modals/edit-approval-permission/edit-approval-permission.component.js

```diff
@@ -0,0 +1,161 @@
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import Modal from '../../modal'
+import Identicon from '../../../ui/identicon'
+import TextField from '../../../ui/text-field'
+import classnames from 'classnames'
+
+export default class EditApprovalPermission extends PureComponent {
+  static propTypes = {
+    hideModal: PropTypes.func.isRequired,
+    selectedIdentity: PropTypes.object,
+    tokenAmount: PropTypes.string,
+    customTokenAmount: PropTypes.string,
+    tokenSymbol: PropTypes.string,
+    tokenBalance: PropTypes.string,
+    setCustomAmount: PropTypes.func,
+  }
+
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  state = {
+    customSpendLimit: this.props.customTokenAmount,
+    selectedOption: this.props.customTokenAmount ? 'custom' : 'unlimited',
+  }
+
+  renderModalContent () {
+    const {
+      hideModal,
+      selectedIdentity,
+      tokenAmount,
+      tokenSymbol,
+      tokenBalance,
+      customTokenAmount,
+    } = this.props
+    const { name, address } = selectedIdentity || {}
+
+    return (
+      <div className="edit-approval-permission">
+        <div className="edit-approval-permission__header">
+          <div className="edit-approval-permission__title">
+            Edit Permission
+          </div>
+          <div
+            className="edit-approval-permission__header__close"
+            onClick={() => hideModal()}
+          />
+        </div>
+        <div className="edit-approval-permission__account-info">
+          <div className="edit-approval-permission__account-info__account">
+            <Identicon
+              address={address}
+              diameter={32}
+            />
+            <div className="edit-approval-permission__account-info__name">{ name }</div>
+            <div>Balance</div>
+          </div>
+          <div className="edit-approval-permission__account-info__balance">
+            {`${tokenBalance} ${tokenSymbol}`}
+          </div>
+        </div>
+        <div className="edit-approval-permission__edit-section">
+          <div className="edit-approval-permission__edit-section__title">
+            Spend limit permission
+          </div>
+          <div className="edit-approval-permission__edit-section__description">
+            Allow Uniswap to withdraw and spend up to the following amount:
```

Comment:
The "Uniswap" here should be replaced with the name for whichever site is requesting approval.
The same goes for the "Spend limit requested by Uniswap" message below